devito/data/utils.py

```diff
@@ -331,3 +309,75 @@ def mpi_index_maps(loc_idx, shape, topology, coords, comm):
         local_si[index] = as_tuple(rnorm_index)
         it.iternext()
     return owners, send, global_si, local_si
+
+
+def flip_idx(idx, decomposition):
+    processed = []
+    for i, j in zip(as_tuple(idx), decomposition):
+        if isinstance(i, slice) and i.step is not None and i.step < 0:
+            if i.start is None:
+                stop = None
+            elif i.start > 0:
+                stop = i.start + 1
+            else:
+                stop = i.start + j.glb_max + 2
+            if i.stop is None:
+                start = None
+            elif i.stop > 0:
+                start = i.stop + 1
+            else:
+                start = i.stop + j.glb_max + 2
+            processed.append(slice(start, stop, -i.step))
+        elif isinstance(i, slice):
+            if i.start is not None and i.start < 0:
+                start = i.start + j.glb_max + 1
+            else:
+                start = i.start
+            if i.stop is not None and i.stop < 0:
+                stop = i.stop + j.glb_max + 1
+            else:
+                stop = i.stop
+            processed.append(slice(start, stop, i.step))
+        else:
+            processed.append(i)
+    return as_tuple(processed)
+
+
+def distributed_data_size(shape, coords, topology, nprocs):
```

Comment:
just to be sure: does this name mean "compute_distributed_data_size" ?

Reply:
It's to compute the 'cumulative distrusted data size' (could maybe call it cum_dist_dat_size? Anyway, alternate name suggestions are welcome!). Quick example:
Rank 0-3 dat shapes are `(2, 2)` and grid is a square.
This function will return:
```
[[(2, 2), (2, 4)],
 [(4, 2), (4, 4)]]
```